### PR TITLE
Emmett pathing

### PR DESCRIPTION
**Pull request by Emmett-Byrne**

### CODE REVIEW

File: ARGO/include/CompEnums.h [hunk 1]
```diff
@@ -18,6 +18,7 @@ enum class ComponentType
 	Command = 14,
 	Tile = 15,
 	FireRate = 16,
+	FlowField,
```
Comment (CathalRedmond) [inline ARGO/include/CompEnums.h]: Might not be too important but you should set this to 17, for consistency

File: ARGO/include/Utilities.h [hunk 2]
```diff
@@ -66,7 +66,12 @@ namespace Utilities
 	const int AUDIO_CHANNELS_PER_PLAYER = 8;
 	const int AUDIO_PLAYERS_FIRE_CHANNELS = AUDIO_MIX_CHANNELS - AUDIO_CHANNELS_PER_PLAYER * 4;
 
-	static const int S_NUMBER_OF_SCREENS = 7; 
+	static const int S_NUMBER_OF_SCREENS = 7;
+
+	//Flow field values
+	const int MAX_FLOW_FIELD_WEIGHT = 20;
+
+	const float RADIANS_TO_DEGREES = 57.2958;
```
Comment (CathalRedmond) [inline ARGO/include/Utilities.h:74]: glm has  two functions radians() and degrees() that converts between radians and degrees

File: ARGO/src/AiSystem.cpp [hunk 3]
```diff
@@ -47,6 +45,42 @@ void AiSystem::update(Entity& t_entity)
...
 {
+	t_aiComponent->setState(AIStates::eWander);
+	Entity* tile = m_levelmanager.findAtPosition(t_posComp->getPos());
+	glm::vec2 seekPosition = glm::vec2(0, 0);
+	if (tile)
+	{
+		FlowFieldComponent* flowFieldComp = static_cast<FlowFieldComponent*>(tile->getComponent(ComponentType::FlowField));
+		if (flowFieldComp && flowFieldComp->getWeight() < Utilities::MAX_FLOW_FIELD_WEIGHT)
+		{
+			if (flowFieldComp->getWeight() == 0)
```
Comment (CathalRedmond) [inline ARGO/src/AiSystem.cpp:56]: this kinda doesnt make sense to have on a seperate line as you are already checking the value above

File: ARGO/src/FlowField.cpp [hunk 4]
```diff
@@ -0,0 +1,2 @@
+#include "stdafx.h"
```
Comment (CathalRedmond) [inline ARGO/src/FlowField.cpp:1]: is flow field.cpp used for anything

Comment (CathalRedmond) [inline ARGO/src/FlowField.cpp:1]: is flow field.cpp used for anything

**Review by CathalRedmond — CHANGES_REQUESTED** [review]

File: ARGO/src/AiSystem.cpp (hunk 3 above)
Comment (Violet-Byrne) [reply ARGO/src/AiSystem.cpp:56]: I'm checking it against 2 different values? it's not the exact same check on both lines

File: ARGO/src/FlowField.cpp (hunk 4 above)
Comment (Violet-Byrne) [reply ARGO/src/FlowField.cpp:1]: Not now but I will likely use it for doing the lighting so I left it there.

Comment (Violet-Byrne) [reply ARGO/src/FlowField.cpp:1]: Not now but I will likely use it for doing the lighting so I left it there.

File: ARGO/src/AiSystem.cpp (hunk 3 above)
Comment (PrzemekTomczyk) [reply ARGO/src/AiSystem.cpp:56]: @CathalRedmond you're a 🐒

**Pull request merged**